internal/pkg/cli/env_delete.go

```diff
@@ -36,6 +136,83 @@ func (opts *DeleteEnvOpts) RecommendedActions() []string {
 	return nil
 }
 
+func (opts *DeleteEnvOpts) initClientsFromRole(roleARN, region string) error {
+	sess, err := session.FromRole(roleARN, region)
+	if err != nil {
+		return fmt.Errorf("get session from role %s and region %s: %w", roleARN, region, err)
+	}
+	opts.rgClient = resourcegroupstaggingapi.New(sess)
+	opts.iamClient = iam.New(sess)
+	opts.deployClient = cloudformation.New(sess)
+	return nil
+}
+
+func (opts *DeleteEnvOpts) shouldDelete(projName, envName string) (bool, error) {
+	if opts.SkipConfirmation {
+		return true, nil
+	}
+
+	shouldDelete, err := opts.prompt.Confirm(fmt.Sprintf(fmtDeleteEnvPrompt, envName, projName), "")
+	if err != nil {
+		return false, fmt.Errorf("prompt for environment deletion: %w", err)
+	}
+	return shouldDelete, nil
+}
+
+// deleteStack returns true if the stack was deleted successfully. Otherwise, returns false.
+func (opts *DeleteEnvOpts) deleteStack() bool {
+	opts.prog.Start(fmt.Sprintf(fmtDeleteEnvStart, opts.env.Name, opts.env.Project))
+	if err := opts.deployClient.DeleteEnvironment(opts.env.Project, opts.env.Name); err != nil {
+		opts.prog.Stop(fmt.Sprintf(fmtDeleteEnvFailed, opts.env.Name, opts.env.Project, err))
+		return false
+	}
+	opts.prog.Stop(fmt.Sprintf(fmtDeleteEnvComplete, opts.env.Name, opts.env.Project))
+	return true
+}
+
+// deleteRoles returns true if the the execution role and manager role were deleted successfully. Otherwise, returns false.
+func (opts *DeleteEnvOpts) deleteRoles() bool {
+	// We must delete the EnvManagerRole last since as it's the assumed role for deletions.
+	roleNames := []string{
+		fmt.Sprintf("%s-%s-%s", opts.env.Project, opts.env.Name, "CFNExecutionRole"),
```

Comment:
Future TODO: Figure out a better way of getting these... Maybe we can save it on the env it self. I think the ENV manager role is actually on the env object.

Reply:
Done!